extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQWorkerTask.java

```diff
@@ -116,6 +119,11 @@ public String getType()
     return TYPE;
   }
 
+  @Override
+  public String getLabel() {
+    return label;
```

Comment:
You can put the label in context. That way we donot explode the constructor args.